### ISSUE TITLE
Pressing Enter to input the IME string also sends Enter to shell, making shell to execute the string as a command

### ISSUE THREAD

**Issue by driver1998** [microsoft/terminal#4041]
# Environment

```none
Windows build number: [run `[Environment]::OSVersion` for powershell, or `ver` for cmd] 10.0.19536.1000
Windows Terminal version (if applicable): 0.7.3451.0

Any other software?
```

# Steps to reproduce
Type something in IME, Microsoft Pinyin IME in this case.
![QQ截图20191222185039](https://user-images.githubusercontent.com/22699485/71320834-00a64580-24ec-11ea-88ef-25824d9d52c7.png)

Use the space bar to select the first candidate, "中文输入法" in this case.

# Expected behavior
Exactly "中文输入法“ is inserted.

# Actual behavior
An additional space is added at the end.
![QQ截图20191222185158](https://user-images.githubusercontent.com/22699485/71320846-351a0180-24ec-11ea-841e-5bc20316f6b6.png)

When I select candidates with number keys, or by clicking them, it works as expected.

Additionally, if I press Enter to input the pinyin string, the additional Enter is also inserted to terminal, making shell to execute that string as a command.
![QQ截图20191222185727](https://user-images.githubusercontent.com/22699485/71320890-f173c780-24ec-11ea-976f-93c04bf06d26.png)
**Press Enter**
![QQ截图20191222185735](https://user-images.githubusercontent.com/22699485/71320893-f9336c00-24ec-11ea-966f-590a37497aa5.png)

**Comment by zadjii-msft** [comment]
So largely, this is a dupe of #3982. However, I'm going to keep this open for the following part of the bug report:

> Additionally, if I press Enter to input the pinyin string, the additional Enter is also inserted to terminal, making shell to execute that string as a command.

That shouldn't happen.

**Comment by skyline75489** [comment]
I'm using 10.0.18363.418 and I can't seem to reproduce this. Maybe it only happens on insider preview?

**Comment by driver1998** [comment]
Looks like both this and #3982 only happens when you use the "new version" of Pinyin IME (aka the TSF3 UWP UI), it does not happen when you use the "old version".

**Comment by driver1998** [comment]
Seems to be fixed on v0.8 and build 19541, validation needed.

**Comment by driver1998** [comment]
This is indeed fixed on v0.8, close this issue.
